bakery/checkers/checkers_test.go

```diff
@@ -559,3 +427,113 @@ func (*CheckersSuite) TestOperationErrorCaveatTest(c *gc.C) {
 		c.Assert(test.caveat.Condition, gc.Matches, test.expectCondition)
 	}
 }
+
+func (*CheckersSuite) TestRegisterNilFuncPanics(c *gc.C) {
+	checker := checkers.New(nil)
```

Comment:
I don't think I've yet seen a test that checks the behaviour of chekers.New(nil)